src/internal/packager/data.go

```diff
@@ -0,0 +1,99 @@
+package packager
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/defenseunicorns/zarf/src/internal/k8s"
+	"github.com/defenseunicorns/zarf/src/internal/message"
+	"github.com/defenseunicorns/zarf/src/internal/utils"
+	"github.com/defenseunicorns/zarf/src/types"
+)
+
+// Wait for the target pod(s) to come up and inject the data into them
+// todo:  this currently requires kubectl but we should have enough k8s work to make this native now
+func handleDataInjection(wg *sync.WaitGroup, data types.ZarfDataInjection, componentPath componentPaths) {
+	message.Debugf("packager.handleDataInjections(%#v, %#v, %#v)", wg, data, componentPath)
+	defer wg.Done()
+
+	injectionCompletionMarker := filepath.Join(componentPath.dataInjections, ".zarf-sync-complete")
+	if err := utils.WriteFile(injectionCompletionMarker, []byte("🦄")); err != nil {
+		message.Errorf(err, "Unable to create the data injection completion marker")
+		return
+	}
+
+	timeout := time.After(15 * time.Minute)
+	for {
+		// delay check 2 seconds
+		time.Sleep(2 * time.Second)
+
+		message.Debugf("Attempting to inject data into %s", data.Target)
+
+		select {
+
+		// on timeout abort
+		case <-timeout:
+			message.Warnf("data injection into target %s timed out\n", data.Target.Namespace)
+			return
+
+		default:
+			source := filepath.Join(componentPath.dataInjections, filepath.Base(data.Target.Path))
+
+			// Wait until the pod we are injecting data into becomes available
+			pods := k8s.WaitForPodsAndContainers(data.Target, true)
+			if len(pods) < 1 {
+				continue
+			}
+
+			// Inject into all the pods
+			for _, pod := range pods {
+				cpPodExec := fmt.Sprintf("tar cz -C %s . | kubectl exec -i -n %s %s -c %s -- tar xzvf - -C %s",
```

Comment:
We talked about making the `z` in `tar xzvf` a flagged input in the `dataInjections:` section of a zarf.yaml right? Do you want me to do that if you're busy with other things?

Reply:
Added now, thx for noting that.